testsuite/module-publish/src/packages/simple/sources/Simple.move

```diff
@@ -402,4 +405,192 @@ module 0xABCD::Simple {
+        let property_types = vector[];
+
+        token::create_tokendata(
+            creator,
+            collection_name,
+            tokendata_name,
+            tokendata_description,
+            maximum,
+            token_uri,
+            creator_address,
+            ROYALTY_POINTS_DENOMINATOR,
+            ROYALTY_POINTS_NUMERATOR,
+            token_mutate_config,
+            property_keys,
+            property_values,
+            property_types
+        )
+    }
+
+
+    /// Mint NFT and store it in a table
+    public entry fun token_v1_mint_and_store_nft_sequential(user: &signer, creator_address: address) acquires MinterConfig {
+        let creator = &get_signer(creator_address);
+
+        // Make the tokendata name unique by appending a ` #` and the current supply + 1
+        let current_supply_opt = token::get_collection_supply(creator_address, string::utf8(COLLECTION_NAME));
+        let index = option::extract(&mut current_supply_opt) + 1;
+        let tokendata_name = build_token_name(string::utf8(TOKEN_NAME), index);
+
+        let tokendata_id = create_token_data(creator, string::utf8(TOKEN_URI), tokendata_name, 1);
+        let token_id = token::mint_token(creator, tokendata_id, 1);
+        let token = token::withdraw_token(creator, token_id, 1);
+        set_token_minted(signer::address_of(user), creator_address, token);
+    }
+
+    /// Mint NFT and transfer it to the user
+    public entry fun token_v1_mint_and_transfer_nft_sequential(user: &signer, creator_address: address) acquires MinterConfig {
+        let creator = &get_signer(creator_address);
+
+        // Make the tokendata name unique by appending a ` #` and the current supply + 1
+        let current_supply_opt = token::get_collection_supply(creator_address, string::utf8(COLLECTION_NAME));
+        let index = option::extract(&mut current_supply_opt) + 1;
+        let tokendata_name = build_token_name(string::utf8(TOKEN_NAME), index);
+
+        let tokendata_id = create_token_data(creator, string::utf8(TOKEN_URI), tokendata_name, 1);
+        let token_id = token::mint_token(creator, tokendata_id, 1);
+        token::direct_transfer(creator, user, token_id, 1);
+    }
+
+    /// Mint NFT and store it in a table
+    public entry fun token_v1_mint_and_store_nft_parallel(user: &signer, creator_address: address) acquires MinterConfig {
+        let creator = &get_signer(creator_address);
+        let tokendata_id = create_token_data(creator, string::utf8(TOKEN_URI), string::utf8(TOKEN_NAME), 1);
+        let token_id = token::mint_token(creator, tokendata_id, 1);
+        let token = token::withdraw_token(creator, token_id, 1);
+        set_token_minted(signer::address_of(user), creator_address, token);
+    }
+
+    /// Mint NFT and transfer it to the user
+    public entry fun token_v1_mint_and_transfer_nft_parallel(user: &signer, creator_address: address) acquires MinterConfig {
+        let creator = &get_signer(creator_address);
```

Comment:
same as above, the minting part can be re-used.

Reply:
Done